soci/soci_index.go

```diff
@@ -216,8 +216,15 @@ type buildConfig struct {
 	artifactsDb         *ArtifactsDb
 	platform            ocispec.Platform
 	optimizations       []Optimization
+	forceGetXAttrs      bool
 }
 
+// Optimization represents an optional optimization to be applied when building the SOCI index
+type Optimization string
+
+// Uncomment the below block when adding optimizations.
+// This is commented out for now to pass CI lint checker.
```

Comment:
Maybe it is best to either remove the code for now or exception the lint checks here if we want to keep this implementation.

Reply:
I think the question to ask is, how likely is it that we find some other form of experimental change? I think it's pretty unlikely, but if it ever happens, creating a new structure for it instead of appending to what's already existing sounds like it could end up being more convoluted than needed. But, having dead code for months, if not indefinitely, sounds like poor coding practice. But it's also not like it takes up a lot of space.

I created a placeholder and excepted the function for now, but very open to just removing it entirely.

Reply:
Yeah I agree with you. I don't have a strong opinion either way.